integration/fuse/src/main/java/alluxio/fuse/AlluxioFuse.java

```diff
@@ -162,55 +163,44 @@ public static FuseUmountable launchFuse(FileSystemContext fsContext, FileSystem
+    if (!Files.exists(mountPath)) {
+      LOG.warn("Mount point on local filesystem does not exist, creating {}", mountPoint);
+      try {
         Files.createDirectories(mountPath);
+      } catch (IOException e) {
+        throw new FailedPreconditionRuntimeException("Failed to create mount point");
       }
+    }
 
-      final boolean debugEnabled = conf.getBoolean(PropertyKey.FUSE_DEBUG_ENABLED);
-      if (conf.getBoolean(PropertyKey.FUSE_JNIFUSE_ENABLED)) {
-        final AlluxioJniFuseFileSystem fuseFs
-            = new AlluxioJniFuseFileSystem(fsContext, fs);
+    final boolean debugEnabled = conf.getBoolean(PropertyKey.FUSE_DEBUG_ENABLED);
+    if (conf.getBoolean(PropertyKey.FUSE_JNIFUSE_ENABLED)) {
+      final AlluxioJniFuseFileSystem fuseFs
+          = new AlluxioJniFuseFileSystem(fsContext, fs);
 
-        FuseSignalHandler fuseSignalHandler = new FuseSignalHandler(fuseFs);
-        Signal.handle(new Signal("TERM"), fuseSignalHandler);
+      FuseSignalHandler fuseSignalHandler = new FuseSignalHandler(fuseFs);
+      Signal.handle(new Signal("TERM"), fuseSignalHandler);
 
-        try {
-          LOG.info("Mounting AlluxioJniFuseFileSystem: mount point=\"{}\", OPTIONS=\"{}\"",
-              mountPoint, String.join(",", optimizedMountOptions));
-          fuseFs.mount(blocking, debugEnabled, optimizedMountOptions);
-          return fuseFs;
-        } catch (FuseException e) {
-          // only try to umount file system when exception occurred.
-          // jni-fuse registers JVM shutdown hook to ensure fs.umount()
-          // will be executed when this process is exiting.
-          String errorMessage = String.format("Failed to mount path %s to mount point %s",
-              targetPath, mountPoint);
-          LOG.error(errorMessage, e);
-          try {
-            fuseFs.umount(true);
-          } catch (FuseException fe) {
-            LOG.error("Failed to unmount Fuse", fe);
-          }
-          throw new IOException(errorMessage, e);
-        }
-      } else {
-        final AlluxioJnrFuseFileSystem fuseFs = new AlluxioJnrFuseFileSystem(fs, conf);
-        try {
-          fuseFs.mount(mountPath, blocking, debugEnabled, optimizedMountOptions);
-          return fuseFs;
-        } catch (ru.serce.jnrfuse.FuseException e) {
-          // only try to umount file system when exception occurred.
-          // jnr-fuse registers JVM shutdown hook to ensure fs.umount()
-          // will be executed when this process is exiting.
-          fuseFs.umount();
-          throw new IOException(String.format("Failed to mount path %s to mount point %s",
-              targetPath, mountPoint), e);
-        }
+      try {
+        LOG.info("Mounting AlluxioJniFuseFileSystem: mount point=\"{}\", OPTIONS=\"{}\"",
+            mountPoint, String.join(",", optimizedMountOptions));
+        fuseFs.mount(blocking, debugEnabled, optimizedMountOptions);
+        return fuseFs;
+      } catch (Throwable t) {
```

Comment:
Is throwable really necessary here? we want to minimize the exception scope so maybe only runtime exception? Or you think we should catch Error as well?

Reply:
Runtime Exception should be enough

Reply:
use throwable before that whatever problem thrown when mount we should try to unmount